src/finch/finch_logic/nodes.py

```diff
@@ -28,27 +52,25 @@ def is_stateful():
         """Determines if the node is stateful."""
         ...
 
-    @classmethod
-    def head(cls):
-        """Returns the head of the node."""
-        return cls
```

Comment:
Let's keep all `head` methods.

Reply:
Done in https://github.com/finch-tensor/finch-tensor-lite/pull/17/commits/56349b0d36760ac0f6f916d9b851a2d6744caddd.